document-generator-accounts/src/main/java/uk/gov/companieshouse/document/generator/accounts/handler/accounts/AccountsHandlerImpl.java

```diff
@@ -83,5 +88,43 @@ private AccountType getAccountType(Accounts accountsData) throws HandlerExceptio
     private String getAccountLink(Accounts accounts, AccountType accountsType) {
         return accounts.getLinks().get(accountsType.getResourceKey());
     }
-    
+
+    /**
+     * Creates the 'data' string in {@link DocumentInfoResponse}.
+     * @param transaction the transaction data
+     * @param abridgedAccountData the abridged accounts data
+     * @return data string in {@link DocumentInfoResponse}
+     */
+    private String createDocumentInfoResponseData(Transaction transaction, AbridgedAccountsApi abridgedAccountData) {
+        JSONObject abridgedAccountJSON = new JSONObject(abridgedAccountData);
+        JSONObject abridgedAccount = new JSONObject();
+        abridgedAccount.put("abridged_account", abridgedAccountJSON);
+        abridgedAccount.put("company_name", transaction.getCompanyName());
+        abridgedAccount.put("company_number", transaction.getCompanyNumber());
+        return abridgedAccount.toString();
+    }
+
+    /**
+     * Creates the {@link DocumentInfoResponse} object
+     * @param transaction transaction data
+     * @param accountType account type
+     * @param abridgedAccountData abridged account data
+     * @return {@link DocumentInfoResponse} object
+     */
+    private DocumentInfoResponse createResponse(Transaction transaction, AccountType accountType, AbridgedAccountsApi abridgedAccountData) {
+        DocumentInfoResponse documentInfoResponse = new DocumentInfoResponse();
+        documentInfoResponse.setData(createDocumentInfoResponseData(transaction, abridgedAccountData));
+        documentInfoResponse.setAssetId(accountType.getAssetId());
+        documentInfoResponse.setTemplateName(accountType.getTemplateName());
+        documentInfoResponse.setLocation(createLocationString(accountType));
+        return documentInfoResponse;
+    }
+
+    private String createLocationString(AccountType accountType) {
+        return String.format("%s/%s/%s", getBucketName(), accountType.getAssetId(), accountType.getUniqueFileName());
+    }
+
+    private String getBucketName() {
+        return bucketName;
```

Comment:
I completely agree, I did make the variable mandatory however, unfortunately the tests fail to run because the `DOC_GEN_ACC_BUCKET_NAME` doesn't exist. I would welcome suggestions to get around that. Normally we have put the variables in the maven config, however because of the modules, I'm wondering if we should be putting `document-generator-accounts` environment variables in the `document-generator`

Reply:
document-generator shouldn't contain the pluggable-code's environment variables. We'll be mixing generic and specific which could cause difficulty in the future updating or removing them. Without having spent time thinking about it, my quick suggestion would be an initialise method in the DocumentInfo interface to allow each implementation to initialise the configs it needs. This will keep the separation and allow failing early (service start-up) if the configs are not available. We would need to discuss with @Mathsh.

Reply:
As discussed offline, we will set the location in `doc-gen-api` and we will only return the path which does not need the `bucketName`